bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingImpl.java

```diff
@@ -81,13 +81,26 @@ public ThingImpl(ThingTypeUID thingTypeUID, String thingId) throws IllegalArgume
     /**
      * @param thingUID
      * @throws IllegalArgumentException
+     * @deprecated use {@link #ThingImpl(ThingTypeUID, ThingUID)} instead.
      */
+    @Deprecated
     public ThingImpl(ThingUID thingUID) throws IllegalArgumentException {
         this.uid = thingUID;
         this.thingTypeUID = new ThingTypeUID(thingUID.getBindingId(), thingUID.getThingTypeId());
```

Comment:
Shouldn't this now throw an IllegalArgumentException in case the thingTypeId is empty?